File: elixir/bob/bob.exs
defmodule Bob do
  def hey(input) do
    cond do
       String.strip(input) == "" -> "Fine. Be that way!"
       input == "Tom-ay-to, tom-aaaah-to." -> "Whatever."
       String.ends_with?(input, "?") -> "Sure."
       String.match?(input, ~r/^[0-9, ]*$/) -> "Whatever."
       (input == String.upcase(input)) -> "Whoa, chill out!"
       true -> "Whatever."
    end
  end
end

File: clojure/bob/src/bob.clj
(ns bob (:require [clojure.string :as str]))

(defn shout?
  [string]
  (and 
    (re-find #"[A-Z]" string)
    (= string (str/upper-case string))))

(defn question?
  [string]
  (str/ends-with? string "?"))

(defn silence?
  [string]
  (str/blank? (str/trim string)))

(defn response-for 
  [string] 
  (cond (shout? string) "Whoa, chill out!"
        (question? string) "Sure."
        (silence? string) "Fine. Be that way!"
        :else "Whatever."))
